File: src/msa/web-service/portalapi/src/main/java/vn/com/kns/portalapi/application/service/category/item/dto/ItemDto.java
package vn.com.kns.portalapi.application.service.category.item.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import vn.com.kns.portalapi.application.service.category.groupItem.dto.GroupItemDto;
import vn.com.kns.portalapi.application.service.category.unit.dto.UnitDto;
import vn.com.kns.portalapi.application.service.supplier.dto.SupplierDto;
import vn.com.kns.portalapi.core.model.BaseEntityDto;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class ItemDto extends BaseEntityDto {
    private String code;
    private String name;
    private String barCode;

    private GroupItemDto groupItem;
    private Long groupItemId;

    private UnitDto unit;
    private Long unitId;
}

File: src/msa/main-service/authw-service/src/app/shared/services/category/item/dto/item-dto.ts
import {GroupItemDto} from '../../group-item/dto/group-item-dto';
import {SupplierDto} from '../../../supplier/dto/supplier-dto';
import {UnitDto} from '../../unit/dto/unit-dto';

export class ItemDto {
  id: number | null;
  code: string;
  name: string;
  price: number = 1;

  unitId: number;
  unit: UnitDto = new UnitDto();

  groupItemId: number;
  groupItem: GroupItemDto = new GroupItemDto();

  supplierId: number;
  supplier: SupplierDto = new SupplierDto();

  // orderMethodId: number;
  // deliveryMethodId: number;
  // returnMethodId: number;
  // storageConditionId: number;

  constructor(id?: number | null, name?: string) {
    this.id = id;
    this.name = name;
  }
}
